src/json.hpp

```diff
@@ -2968,20 +2994,35 @@ class parser
     @throw parse_error.101 in case of an unexpected token
     @throw parse_error.102 if to_unicode fails or surrogate error
     @throw parse_error.103 if to_unicode fails
+    @throw out_of_range 406 if float not finite
     */
     void parse(const bool strict, BasicJsonType& result)
     {
+        std::unique_ptr<exception> ex;
+        parse(strict, result, ex);
+        try_throw_any_exception(ex.get());
```

Comment:
`std::rethrow_exception` could be used here (assuming we replace `ex` with a `std::exception_ptr`.

Reply:
ah interesting thanks! I actually have never heard of `std::exception_ptr` before. Unfortunately it looks like it isn't very useful in this situation. Specifically I want to be able to get at the message on why the parse failed.